junction/proposals/templatetags/proposal_tags.py

```diff
@@ -0,0 +1,16 @@
+from django import template
+register = template.Library()
+from junction.proposals import models
+
+
+@register.assignment_tag
+def get_comment_vote_value(comment_id, user):
+
+    comment_note_value = 0
+    vote = models.ProposalCommentVote.objects.filter(proposal_comment_id=comment_id, voter=user)
+    if vote.exists():
+        if vote[0].up_vote is True:
+            comment_note_value = 1
+        else:
+            comment_note_value = -1
+    return comment_note_value
```

Comment:
this looks better. There are two more things you can do.

1. move all the logic from here to a function `get_comment_vote_value_for_user(user, comment)` in `proposals/services.py`

2. in the `get_comment_vote_value_for_user` function, get the comment value as it is, but then add caching layer for user and comment.

```
from django.core.cache import cache

def get_comment_vote_value_for_user(user, comment):
  cache_key = "cvote:{user_id}:{comment_id}".format(user_id=user.id, comment_id=comment.id)
  vote_value = cache.get(cache_key)
  if vote_value is not None:
    return vote_value

  # ... logic to calculate vote_value here

  cache.set(cache_key, vote_value)
  return vote_value
```

Let me know if all sounds good?